base-containers/base/inginious_container_api/run_student.py

```diff
@@ -7,18 +7,22 @@
 import signal
 import socket
 import tempfile
-
+import asyncio
+import zmq.asyncio
 import msgpack
 import zmq
+import struct
+
+from inginious_container_api.utils import User
 
 def run_student(cmd, container=None,
         time_limit=0, hard_time_limit=0,
         memory_limit=0, share_network=False,
         working_dir=None, stdin=None, stdout=None, stderr=None,
-        signal_handler_callback=None):
+        signal_handler_callback=None, ssh=False, run_as_root=False):
     """
     Run a command inside a student container
-    :param cmd: command to be ran (as a string, with parameters)
+    :param cmd: command to be ran (as a string, with parameters). If ssh is set to True, this command will be run before launching the ssh server.
```

Comment:
Is it possible to have a process that continues to work once the SSH server is started?
(that is, to detach a process that would continue to work in background).

It would be nice to document a small example explaining how to do that.

Reply:
Right now with the curent implementation, the command is started in a thread p and we p.wait() before launching the ssh server...
But actually, it would be easy to add an argument to specify that we want the command to continue in its thread and to stop only when the exercice is finished. Thus in this case, we would not p.wait() directly but instead we would start the ssh server and maybe p.wait or p.kill at the end before exiting the container.

Do you think such an argument would be useful ?
--continue: if you want your command (probably a script) to continue running in background while the student has ssh access.